src/components/attendee/VideoComms/types.ts

```diff
@@ -0,0 +1,97 @@
+import Twilio from "twilio-video";
+
+import { Branded } from "utils/types";
+
+export type TrackId = Branded<string, "TrackId">;
+export type ChannelId = Branded<string, "ChannelId">;
+
+export enum VideoCommsStatus {
+  Disconnected = "DISCONNECTED",
+  Connecting = "CONNECTING",
+  Connected = "CONNECTED",
+  Errored = "ERRORED",
+}
+
+export enum VideoSource {
+  Webcam = "WEBCAM",
+  Screenshare = "SCREENSHARE",
+}
+
+export type VideoTrack = {
+  id: TrackId;
+  attach: (el: HTMLVideoElement) => void;
+  detach: () => void;
+  kind: "video";
+  twilioTrack: Twilio.VideoTrack;
+  sourceType: VideoSource;
+  enabled: boolean;
+};
+
+export type AudioTrack = {
+  id: TrackId;
+  attach: (el: HTMLAudioElement) => void;
+  detach: () => void;
+  kind: "audio";
+  twilioTrack: Twilio.AudioTrack;
+  enabled: boolean;
+};
+
+export type Track = VideoTrack | AudioTrack;
+
+export interface Participant {
+  /**
+   * ID of the participant - used for communication with the video platform
+   */
+  id: string;
+
+  /**
+   * ID of the participant from the point of view of Sparkle. e.g. userId
+   */
+  sparkleId: string;
```

Comment:
Still, weird there is `id` and `sparkleId` instead of `id` and `twillioId` considering every other object (not sure if this one is stored in DB currently) follows the latter style.

Reply:
I'm renaming `id` to `twilioId` and keeping `sparkleId` so there's no chance of accidentally using the wrong one.

Reply:
Resolved in 03a5d4919